src/app/clients/clients-view/view-share-account/view-share-account.component.ts

```diff
@@ -0,0 +1,26 @@
+import { Component, OnInit } from '@angular/core';
+import { ActivatedRoute } from '@angular/router';
+
+@Component({
+  selector: 'mifosx-view-share-account',
+  templateUrl: './view-share-account.component.html',
+  styleUrls: ['./view-share-account.component.scss']
+})
+export class ViewShareAccountComponent implements OnInit {
+  chargesColumns: string[] = ['Name', 'Fee/Penalty', 'Payment due at', 'Calculation Type', 'Due', 'Paid', 'Waived', 'Outstanding'];
+  transactionColumns: string[] = ['Transaction Date', 'Transaction Type',	'Total Shares', 'Purchased/Redeemed Price', 'Charge Amount', 'Amount Received/Returned'];
+  dividendColumns: string[] = ['Transaction Date', 'Amount', 'Transaction Reference', 'Status'];
+  shareAccount: any;
+
+  constructor(
+    private route: ActivatedRoute,
+  ) {
+    this.route.data.subscribe((data: {shareAccountsData: any}) => {
```

Comment:
try to do initialisation logic in ngOnInit lifecycle hook

Reply:
Will do.